test/e2e/commatrix_suite_test.go

```diff
@@ -108,8 +152,93 @@ var _ = AfterSuite(func() {
+
+		By(fmt.Sprintf("get documented commatrix version %s", clusterVersion))
+		// get documented commatrix from URL
+		resp, err := http.Get(strings.Replace(docCommatrixBaseURL, "VERSION", clusterVersion, 1))
+		Expect(err).ToNot(HaveOccurred())
+		defer resp.Body.Close()
+		// if response status code equals to "status not found", compare generated commatrix to the master documented commatrix
+		if resp.StatusCode == http.StatusNotFound {
+			resp, err = http.Get(strings.Replace(docCommatrixBaseURL, "enterprise-VERSION", "main", 1))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(resp.StatusCode).ToNot(Equal(http.StatusNotFound))
+		}
+
+		By("write documented commatrix to artifact file")
+		docCommatrixContent, err := io.ReadAll(resp.Body)
+		Expect(err).ToNot(HaveOccurred())
+		docFilePath := filepath.Join(artifactsDir, "doc-commatrix.csv")
+		err = os.WriteFile(docFilePath, docCommatrixContent, 0644)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Filter documented commatrix for diff generation")
+		// get origin documented commatrix details
+		docComMatrixCreator, err := commatrixcreator.New(epExporter, docFilePath, types.FormatCSV, infra, deployment)
+		Expect(err).ToNot(HaveOccurred())
+		docComDetailsList, err := docComMatrixCreator.GetComDetailsListFromFile()
+		Expect(err).ToNot(HaveOccurred())
+
+		if isSNO {
+			// Exclude all worker nodes static entries.
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: docComDetailsList}, "worker")
+			// Exclude static entries of standard deployment type.
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.StandardStaticEntries}, "master")
+		} else {
+			// Exclude specific master entries (see StandardExcludedMasterComDetails var description)
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: StandardExcludedMasterComDetails}, "master")
+		}
+
+		// if cluster is running on BM exclude Cloud static entries in diff generation
+		// else cluster is running on Cloud and exclude BM static entries in diff generation.
+		if isBM {
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.CloudStaticEntriesWorker}, "worker")
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.CloudStaticEntriesMaster}, "master")
+		} else {
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.BaremetalStaticEntriesWorker}, "worker")
+			docComDetailsList = excludeStaticEntriesWithGivenNodeRole(docComDetailsList, &types.ComMatrix{Matrix: types.BaremetalStaticEntriesMaster}, "master")
+		}
+		docComMatrix := &types.ComMatrix{Matrix: docComDetailsList}
+
+		By("generating diff between matrices for testing purposes")
+		diff := matrixdiff.Generate(commatrix, docComMatrix)
```

Comment:
Maybe to make it more readable worth changing the order here, so docs is first, to align with the name of the file next and to generate UniquePrimary first

Reply:
but as we discussed we want to add `+` where we need to add to the doc, and `-` where we need to remove from the doc, so for that the generated commatrix should be the primary.